src/daemon/rpmostree-sysroot-upgrader.c

```diff
@@ -1016,6 +1017,22 @@ rpmostree_sysroot_upgrader_deploy (RpmOstreeSysrootUpgrader *self,
                                    cancellable, error))
     return FALSE;
 
+  /* Also do a sanitycheck even if there's no local mutation; it's basically free
+   * and might save someone in the future.  The RPMOSTREE_SKIP_SANITYCHECK
+   * environment variable is just used by test-basic.sh currently.
+   */
+  if (!self->final_revision && !getenv ("RPMOSTREE_SKIP_SANITYCHECK"))
```

Comment:
Shouldn't we put the `getenv` check in `rpmostree_deployment_sanitycheck()` itself?